src/test/scala/org/ergoplatform/ErgoScriptPredefSpec.scala

```diff
@@ -1,20 +1,198 @@
 package org.ergoplatform
 
 import org.ergoplatform.ErgoAddressEncoder.TestnetNetworkPrefix
+import org.ergoplatform.ErgoBox.R4
+import org.ergoplatform.mining.emission.EmissionRules
+import org.ergoplatform.settings.MonetarySettings
+import org.scalacheck.Gen
 import scorex.crypto.hash.{Blake2b256, Digest32}
-import sigmastate.AvlTreeData
+import scorex.util.Random
+import sigmastate.Values.{ByteArrayConstant, CollectionConstant, IntConstant, SigmaPropConstant, Value}
+import sigmastate._
+import sigmastate.basics.DLogProtocol.{DLogProverInput, ProveDlog}
 import sigmastate.helpers.{ErgoLikeTestProvingInterpreter, SigmaTestingCommons}
 import sigmastate.interpreter.Interpreter.{ScriptNameProp, emptyEnv}
 import sigmastate.interpreter.{ContextExtension, ProverResult}
-import sigmastate.utxo.ErgoLikeTestInterpreter
+import sigmastate.lang.Terms.ValueOps
+import sigmastate.serialization.ValueSerializer
+import sigmastate.utxo.{ByIndex, ErgoLikeTestInterpreter, ExtractCreationInfo, SelectField}
 
 import scala.util.Try
 
 class ErgoScriptPredefSpec extends SigmaTestingCommons {
   implicit lazy val IR = new TestingIRContext {
     override val okPrintEvaluatedEntries: Boolean = false
   }
-  val emptyProverResult: ProverResult = ProverResult(Array.emptyByteArray, ContextExtension.empty)
+  private val emptyProverResult: ProverResult = ProverResult(Array.emptyByteArray, ContextExtension.empty)
+  private val settings = MonetarySettings(30 * 2 * 24 * 365, 90 * 24 * 30, 75L * EmissionRules.CoinsInOneErgo,
+    3L * EmissionRules.CoinsInOneErgo, 720, 75L * EmissionRules.CoinsInOneErgo / 10)
+  private val emission = new EmissionRules(settings)
+
+  ignore("boxCreationHeight") {
```

Comment:
I wrote about it directly to you - this test fails with `Invalid primitive in Cost function: MethodCall(s39, scalan.Base$Exp special.sigma.SigmaDsl$CostModel.SelectField(), [], true)` error.
Please check it if it is a bug in sigma, because the same script works fine as a part a bigger one (e.g. emissionBoxProp have this line)

Reply:
@aslesarenko @catena2w fixed in https://github.com/ScorexFoundation/sigmastate-interpreter/pull/371